### ISSUE TITLE
WasmPlugin causes pod to fail to start when failStrategy = FAIL_OPEN

### ISSUE THREAD

**Issue by MaYuan-02** [istio/istio#53279]
(This is used to request new product features, please visit <https://github.com/istio/istio/discussions> for questions on using Istio)

**Describe the feature request**

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

The current behavior of the wasmplugin's failStrategy is unreasonable. When I set the failStrategy to FAIL_OPEN, I expect the requests to proceed normally when this plugin is unavailable. However, I encountered a situation where I configured an incorrect image address, causing the plugin to fail to fetch properly, which directly impacted the pod's startup.

##  root cause:
* The current logic sets a fixed InitialFetchTimeout = 0 in the httpfilter. This configuration will definitely lead to a pod startup failure when the image pull fails. 
* In the pilot-agent, if the image pull fails and the failStrategy is set to FAIL_OPEN, an ALLOW_ALL RBAC filter is applied.

## My expected behavior is:

* As long as the failStrategy is set to FAIL_CLOSE, any error from the wasm plugin should result in request rejection.

* As long as the failStrategy is set to FAIL_OPEN, any error from the wasm plugin should not affect my requests.

Are there any plans to fix this?

**Comment by MaYuan-02** [comment]
Envoy's `ExtensionConfigSource` already provides the appropriate fields. 
```json
{
  "config_source": {...},
  "default_config": {...},
  "apply_default_config_without_warming": ...,
  "type_urls": []
}
```
We can configure an ALLOW_ALL or DENY_ALL RBAC filter in the `default_config` based on the failStrategy to achieve this functionality. Then, we set `apply_default_config_without_warming` to true. This configuration aligns better with the intended purpose of the failStrategy.

**Comment by howardjohn** [comment]
How does the existing behavior not meet those requirements? Doesn't sending the allow_any rbac on failure mean "any error from the wasm plugin should not affect my requests."?

`apply_default_config_without_warming` is not the same as FAIL_OPEN, since even on success there is a race before the config is loaded

**Comment by MaYuan-02** [comment]
@howardjohn 😊 Thanks for your reply.
I'm sorry for the earlier miscommunication. The current behavior of FAIL_OPEN is as expected. What I want to express is that FAIL_CLOSE does not meet expectations.

* When the pod starts, if my plugin is set to FAIL_CLOSE, the pod cannot start normally, which is too STRICT. I only want to intercept INBOUND requests, but at this point, the entire pod is non-functional, and other OUTBOUND requests cannot work either.

* While the pod is running, if I create a FAIL_CLOSE plugin, but the wasm image pull fails, no ECDS configuration will be applied, yet my requests can still succeed. If this is an authentication/authorization plugin, this is also not the expected behavior.

**Comment by MaYuan-02** [comment]
@ramaraochavali @howardjohn any comments about this?

**Comment by ramaraochavali** [comment]
I think it makes sense to apply a default DENY all in this case so that rest of the proxy works correctly. @kyessenov also for his inputs

**Comment by howardjohn** [comment]
cc @igsong

**Comment by kyessenov** [comment]
Yeah, it should be per-listener impact, not stall the whole proxy.

**Comment by MaYuan-02** [comment]
#53363 
In this PR, I added the logic to issue a deny-all RBAC filter and fixed the previously allow-all network filter. If you can help review it, I would be very grateful.
